shivyc/parser/declaration.py

```diff
@@ -303,8 +310,32 @@ def parse_struct_spec(index):
     return decl_nodes.Struct(name, members, r), index
 
 
-def parse_struct_members(index):
-    """Parse the list of members of a struct as a list of Root nodes.
+def parse_union_spec(index):
```

Comment:
If I'm not mistaken, `parse_union_spec` and `parse_struct_spec` are almost identical, except for the final return statement. Maybe create a common helper function `_parse_struct_union_spec(index, node)` that takes either `decl_nodes.Union` or `decl_nodes.Struct` as the second argument and contains all the logic currently in the `parse_union_spec` and `parse_struct_spec` functions. Then, you could simply define:
```python
def parse_union_spec(index):
  return _parse_struct_union_spec(index, decl_nodes.Union)
```
and analogously for `parse_struct_spec`. Let me know if this doesn't make sense and I can clarify my comment.